src/dex/fluid-dex/fluid-dex-pool.ts

```diff
@@ -26,19 +33,138 @@ export class FluidDexEventPool extends StatefulEventSubscriber<FluidDexPoolState
     protected network: number,
     protected dexHelper: IDexHelper,
     logger: Logger,
-    protected fluidDexIface = new Interface(FluidDexABI), // TODO: add any additional params required for event subscriber
+    protected liquidityIface = new Interface(LiquidityABI), // TODO: add any additional params required for event subscriber
   ) {
     // TODO: Add pool name
     super(parentName, pool.id, dexHelper, logger);
 
     // TODO: make logDecoder decode logs that
-    this.logDecoder = (log: Log) => this.fluidDexIface.parseLog(log);
-    this.addressesSubscribed = [pool.address, pool.token0, pool.token1];
+    this.logDecoder = (log: Log) => this.liquidityIface.parseLog(log);
+    this.addressesSubscribed = [pool.liquidityUserModule];
 
     // Add handlers
-    // this.handlers['Swap'] = this.handleSwap.bind(this);
+    this.handlers['LogOperate'] = this.handleOperate.bind(this);
   }
 
+  /**
+   * Handle a trade rate change on the pool.
+   */
+  async handleOperate(
+    event: any,
+    state: DeepReadonly<FluidDexPoolState>,
+    log: Readonly<Log>,
+  ): Promise<DeepReadonly<FluidDexPoolState> | null> {
+    const ResolverAbi = new Interface(ResolverABI);
+    if (
+      !(
+        event.args.user in
+        [
+          this.pool.address,
```

Comment:
only pool.address is needed here

the other ones you listed are only ever delegateCalled by the pool main contract logic

Reply:
also we can remove those from config then

Reply:
done👍